src/02-common-principles.md

```diff
@@ -383,6 +383,286 @@ In particular, if a BIDS dataset contains a `derivatives/` sub-directory,
+or a neuroimaging dataset.
+It may just as well be a (sub-)directory containing one or more arbitrary files.
+
+In the case where a derivatives dataset is nested under a raw dataset and both have a `dataset_description.json` file,
+the BIDS URIs within the nested derivatives dataset MUST be resolved with respect to `/derivatives/dataset_description.json`,
+and the BIDS URIs for the raw dataset (excluding `/derivatives`) MUST be resolved with respect to `/dataset_description.json`.
+
+However if a `derivatives/` *folder* is nested under a raw dataset and only the raw dataset has a `dataset_description.json` file,
+that `derivatives/` *folder* is not considered a BIDS dataset (see [Storage of Derivated Datasets](#storage-of-derived-datasets)),
+and all BIDS URIs MUST be resolved with respect to the `/dataset_description.json` file.
+The latter situation is then comparable to referring to data in a `sourcedata/` or `code/` folder.
+
+(Assuming that in the two above examples `/` refers to the root of the BIDS raw dataset with a nested derivatives dataset/folder.)
+
+### Refer to a file within a dataset
+
+To link for example to a derivative file that is stored in a nested `derivatives/`
+directory within a dataset:
+`bids:deriv1:/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`
+
+```json
+{
+    "DatasetLinks": {
+        "deriv1": "file://derivatives/derivative1"
+    }
+
+}
+```
+
+Alternatively, we can also make use of the reserved `<dataset-name>`: `""` (an empty string),
+which always points to the root of the current dataset, to point to the same file using the following syntax:
+`bids::/derivatives/derivative1/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`
+
+Thus, for this specific case there are two ways to refer to the same file:
+
+1.  `bids:deriv1:/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`,
+    with `deriv1` being specified within `dataset_description.DatasetLinks`,
+    as shown in the example above.
+
+1.  `bids::/derivatives/derivative1/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`,
+    with `""` (an empty string) NOT being a key in `dataset_description.DatasetLinks`,
+    because it is a reserved value that always points to the root of the current dataset.
+
+**If *all* BIDS URIs in a dataset *only* use an empty string (`""`) as a `<dataset-name>`,**
+**the `DatasetLinks` metadata MAY not be specified.**
+
+### Refer to a file outside of a dataset but on the same host
+
+It is RECOMMENDED to refer to files **within** a dataset (as described in the section above),
+or on a **remote** location (as described in the section below).
+However sometimes it may be convenient to refer to files that are outside of a given dataset but on the same host.
+BIDS URIs allow for specifying such locations,
+but such specifications are by definition not portable in that the BIDS URIs break when the host changes.
+When sharing a BIDS dataset, the dataset curator MUST make sure that all BIDS URIs are portable.
```

Comment:
What does this mean for the validator? Should the validator reject relative paths, have a relative-path rejecting mode that something like OpenNeuro can use, or something else?